src/Diagnostics.DataProviders/ChangeAnalysisClient.cs

```diff
@@ -300,12 +301,12 @@ private async Task<string> PrepareAndSendRequest(string requestUri, object postB
 
             // Add required headers.
             requestMessage.Headers.Add("Authorization", authToken);
-            requestMessage.Headers.Add("x-ms-client-object-id", clientObjectIdHeader);
+            requestMessage.Headers.Add(ClientObjectIdHeader, clientObjectIdHeader);
 
             // For requests coming from Diagnose and Solve, add x-ms-client-principal-name header.
             if (!string.IsNullOrWhiteSpace(clientPrincipalNameHeader))
             {
-               requestMessage.Headers.Add("x-ms-client-principal-name", clientPrincipalNameHeader);
+                requestMessage.Headers.Add(ClientPrincipalNameHeader, clientPrincipalNameHeader);
```

Comment:
@rekhaswaminathan  - Just wanted to make sure that we have fixed this header name in the variable also? This header is used by Azure Tools team.

Reply:
nvmind. I found the class below while reviewing the PR:

/// <summary>
        /// Client Principal Name header.
        /// </summary>
        public static readonly string ClientPrincipalNameHeader = "x-ms-client-principal-name";

Reply:
Yeah I checked it, its correct in variable.